tests/test_checkers.py

```diff
@@ -890,6 +890,10 @@ def test_union_typevar(self):
         T = TypeVar("T", bound=Parent)
         check_type(Child, Type[T])
 
+    def test_generic_aliase(self):
+        if sys.version_info >= (3, 9):
+            check_type(dict[str, str], type)
```

Comment:
Wouldn't this be better?
```suggestion
    def test_generic_aliases(self):
        check_type(Dict[str, str], Type[Any])
```

Reply:
This check raises another problem: `Dict[str, str]` is of type `typing._GenericAlias`, not `types.GenericAlias`. It cannot be covered by this PR.

I'm not using generic types from typing, and didn't notice this problem.

Reply:
The typing module comes in python 3.5. The github repo for cpython has tags from 3.8.

From there, I see that in 3.8, `Dict` and `Dict[str, str]` are all of type `typing._GenericAlias`. From 3.9, `Dict` has type `typing._SpecialGenericAlias` and `Dict[str, str]` is of type `typing._GenerecAlias`, both are inherited from `typing._BaseGenericAlias`. `_BaseGenericAlias` does not exist in python 3.8. And as the name shows, they are not part of the public interface of `typing`. I think I can only check for these types with:

```python
isinstance(Dict[str, str], type(Type[Any])) # typing._GenericAlias
isinstance(Dict, type(Type)) # typing._SpecialGenericAlias
```

I think all of them should be checked and tested.

Also, this might not work for older python version. I found in python 3.6, `Dict` inherited from `dict` (and some other classes), and `_GenericAlias` does not exit. I did not find a python 3.5.

I only run the actual test in python 3.11.

Reply:
Any Python version older than 3.8 is irrelevant here. And even if those private types from `typing` aren't public, they can still be used as long as we confine that use to below a maximum Python version (3.8?).

Reply:
I've tweaked the checks and the tests somewhat. I think it's ripe for merging now.